### ISSUE TITLE
Dropping characters on fast device->host communication

### ISSUE THREAD

**Issue by obra** [bjc/ArduinoCore-GD32#10]
This looks similar to #8 but is during communications in the other direction.

Demo sketch:

```
void setup() {
  Serial.begin(9600);
}

void loop() {
 for(auto i=0 ;i<1024; i++) 
  Serial.print("0");
 for(auto i=0; i<1024; i++) 
  Serial.print("1");
 for(auto i=0; i<1024; i++) 
  Serial.print("2");
 for(auto i=0 ;i<1024; i++) 
  Serial.print("3");
 for(auto i=0 ;i<1024; i++) 
  Serial.print("4");
 for(auto i=0; i<1024; i++) 
  Serial.print("5");
 for(auto i=0; i<1024; i++) 
  Serial.print("6");
 for(auto i=0 ;i<1024; i++) 
  Serial.print("7");
 for(auto i=0 ;i<1024; i++) 
  Serial.print("8");
 for(auto i=0 ;i<1024; i++) 
  Serial.print("9");
  Serial.println();
  delay(2000);

}
```

Trailing edge of output:

![image](https://user-images.githubusercontent.com/45416/160701408-59417401-58f9-4b23-b463-3421fc29cf22.png)

**Comment by obra** [comment]
It looks like this is a buffer overfilling issue *somewhere* in the pipeline.
This egregious hack gets me reliable writes:
```
void setup() {
  Serial.begin(9600);
}

void myprint(char *character) {
  Serial.write(character);
  delayMicroseconds(40); // A delay of 25 microseconds still saw dropped characters
}

void loop() {
 for(auto i=0 ;i<1024; i++) 
  myprint("0");
 for(auto i=0; i<1024; i++) 
  myprint("1");
 for(auto i=0; i<1024; i++) 
  myprint("2");
 for(auto i=0 ;i<1024; i++) 
  myprint("3");
 for(auto i=0 ;i<1024; i++) 
  myprint("4");
 for(auto i=0; i<1024; i++) 
  myprint("5");
 for(auto i=0; i<1024; i++) 
  myprint("6");
 for(auto i=0 ;i<1024; i++) 
  myprint("7");
 for(auto i=0 ;i<1024; i++) 
  myprint("8");
 for(auto i=0 ;i<1024; i++) 
  myprint("9");
  Serial.println();

}
```

**Comment by bjc** [comment]
Fixed in c4629d85e5740c4b8f4af9a5dde3915372022d67

As I recall, the issue was with heavy flushing causing the macOS host driver for CDC ACM to become confused and start dropping characters and/or inserting NUL bytes.
